power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -29,6 +29,7 @@ inline void iterate_component_sequence(Func&& func, ForwardIterator begin, Forwa
         // get component directly using sequence id
         func(*it, sequence_idx[seq]);
     }
+    (void)std::forward<Func>(func);
```

Comment:
why this?

Reply:
this function captures `func` by perfect forwarding. It is not allowed by the standard to not forward a forwarded argument, or move a moved argument.

Reply:
Same here for https://github.com/PowerGridModel/power-grid-model/pull/882#discussion_r1938948404.

`func` should be defined as pass by value in the function argument.